### DISCUSSION TITLE
cbc parameters setting

### DISCUSSION THREAD

**Discussion by mingcaixiao** [coin-or/Cbc#455] [Q&A] [answered]
hello everyone!
I am a new user of cbc.when i use cbc stand-alone command-line program and use cbc as library in C++,I found that for same MIP problem,cbc stand-alone program behave better than using it in C++ .In stand-alone,it cosumed less time to find a good solution.
I refer to /examples/minimum.cpp to read mps file and solve it.
Is defualt parameters setting of C++ API different from stand-alone program?If it is,how to set parameter in C++ same with cbc stand-alone program?

**Comment by jhmgoossens** [comment, marked as answer]
The trick is to call Cbc in c++ using the CbcMain0 and CbcMain1 as used around line 500 of [Driver5.cpp](https://github.com/coin-or/Cbc/blob/master/examples/driver5.cpp) 
I believe this way to call will change in the near future.
I hope this answers your question?

**Reply by mingcaixiao** [reply]
Thank you very much!
But,for cbc 2.10.5, using of CbcMain1() in [driver5.cpp](https://github.com/coin-or/Cbc/blob/master/examples/driver5.cpp) can‘t match function parameters correctly.It seems the CbcMain1() in new version of cbc has changed. [driver6](https://github.com/coin-or/Cbc/blob/master/examples/driver6.cpp) did works.

**Reply by jhmgoossens** [reply]
Indeed, the function signature has changed since 2.10.5. I believe driver5 in Master has the current signature, but probably  driver6 wasn’t updated yet. Glad you got it to work.
